tests/e2e/scale/test_pvc_metrics.py

```diff
@@ -0,0 +1,174 @@
+from ocs_ci.ocs.metrics import collect_pod_metrics
+
+log = logging.getLogger(__name__)
+
+
+@scale
+#
+# NOTE: This code class was cannibalized from test_pvc_creation_deletion_scale.py
+#       This is a WIP branch which I would like to clean up by not duplicating this code.
+#
+class TestScaleAndRecordMetrics(E2ETest):
+    """
+    Base class for testing metrics recording
+    """
+
+    @pytest.fixture()
+    def namespace(self, project_factory):
+        """
+        Create a new project
+        """
+        self.start_time = default_timer()
+        proj_obj = project_factory()
+        self.namespace = proj_obj.namespace
+
+    @pytest.mark.parametrize(
+        argnames=["access_mode", "interface"],
+        argvalues=[
+            pytest.param(
+                *[constants.ACCESS_MODE_RWO, constants.CEPHBLOCKPOOL],
+                marks=pytest.mark.polarion_id("OCS-1225"),
+            ),
+            pytest.param(
+                *[constants.ACCESS_MODE_RWX, constants.CEPHBLOCKPOOL],
+                marks=pytest.mark.polarion_id("OCS-2010"),
+            ),
+            pytest.param(
+                *[constants.ACCESS_MODE_RWX, constants.CEPHFS_INTERFACE],
+                marks=pytest.mark.polarion_id("OCS-2008"),
+            ),
+        ],
+    )
+    @pytest.mark.usefixtures(namespace.__name__)
+    def test_multiple_pvc_creation_deletion_scale(
+        self, es, namespace, tmp_path, access_mode, interface
+    ):
+        """
+        Measuring PVC creation time while scaling PVC
+        Measure PVC deletion time after creation test
+        """
+        crd_data = templating.load_yaml(constants.OSD_SCALE_BENCHMARK_YAML)
+        our_uuid = uuid4().hex
+        self.elastic_info = ElasticData(our_uuid, crd_data)
+        self.elastic_info.es_connect()
+        metrics = collect_pod_metrics()
+        logging.info("Beginning of run METRICS collection")
+        logging.info(metrics)
+        scale_pvc_count = scale_lib.get_max_pvc_count()
+        log.info(f"Start creating {access_mode}-{interface} {scale_pvc_count} PVC")
+        if interface == constants.CEPHBLOCKPOOL:
+            sc_name = constants.DEFAULT_STORAGECLASS_RBD
+        elif interface == constants.CEPHFS_INTERFACE:
+            sc_name = constants.DEFAULT_STORAGECLASS_CEPHFS
+
+        # Get pvc_dict_list, append all the pvc.yaml dict to pvc_dict_list
+        pvc_dict_list1 = scale_lib.construct_pvc_creation_yaml_bulk_for_kube_job(
+            no_of_pvc=int(scale_pvc_count / 2), access_mode=access_mode, sc_name=sc_name
+        )
+        pvc_dict_list2 = scale_lib.construct_pvc_creation_yaml_bulk_for_kube_job(
+            no_of_pvc=int(scale_pvc_count / 2), access_mode=access_mode, sc_name=sc_name
+        )
+
+        # There is 2 kube_job to reduce the load, observed time_out problems
+        # during delete process of single kube_job and heavy load.
+        job_file1 = ObjectConfFile(
```

Comment:
I think you can make a list out of n number of jobs instead of initializing individual objects.  like:
```python
njobs=2
job_file_list=[]

for i in njobs:
   job_file_list[i] = ObjectConfFile(
                            name=f"job_profile_{i}",
                            obj_dict_list=pvc_dict_list[i]    # Assuming pvc_dict_list is a list with pvc dictionaries
                            project=self.namespace,
                            tmp_path=tmp_path,
             )

for job in job_file_name:
   job.create()
```

In this way it makes the testcase flexible to extend it for N numbers and reduces the lines. WDYT ?

Reply:
This code was cannibalized from test_pvc_creation_deletion.py and I will not be using this in my final version.